### PR TITLE
fix(hook): prefix var from outgoing hook

### PR DESCRIPTION
**Pull request by yesnault**
This prefix : 
 - allow user to know exactly the context of the root workflow
 - allow user to choose between the default payload of the child or the variable received from the hook
 - will avoid some troubles with .git.values on child workflow

Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

**Comment by fsamin** [comment]
so it is impossible to override the default payload of the root child workflow from a parent workflow :(
i'm afraid a disagree. 
The goal of default payload is to be overriden.

**Comment by yesnault** [comment]
ook. prefix is now only for "cds.".. and "git.".. variables.

capture d ecran 2018-11-23 a 09 29 56

**Comment by sguiheux** [comment]
This is weird:   allow user to choose between the default payload of the child or the variable received from the hook

The only payload that matter is the payload sent by the hook

**Comment by yesnault** [comment]
> allow user to choose between the default payload of the child or the variable received from the hook

should be good with the last commit.

Prefix only the variable beginning with cds. and git.

**Comment by yesnault** [comment]
-> payload is now explicitly choose by user
-> parent.name / parent.key / parent.run -> var added (not in payload) in build params

File: engine/api/workflow/process_parameters.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func getNodeJobRunParameters(db gorp.SqlExecutor, j sdk.Job, run *sdk.WorkflowNo
 }
 
 // GetNodeBuildParameters returns build parameters with default values for cds.version, cds.run, cds.run.number, cds.run.subnumber
-func GetNodeBuildParameters(proj *sdk.Project, w *sdk.Workflow, runContext nodeRunContext, pipelineParameters []sdk.Parameter, payload interface{}) ([]sdk.Parameter, error) {
+func GetNodeBuildParameters(proj *sdk.Project, w *sdk.Workflow, runContext nodeRunContext, pipelineParameters []sdk.Parameter, payload interface{}, hookEvent *sdk.WorkflowNodeRunHookEvent) ([]sdk.Parameter, error) {
```
Comment (sguiheux) [inline engine/api/workflow/process_parameters.go]: maybe hookEvent must be part of runContext struct ?

File: engine/api/workflow/process_parameters.go [hunk 2]
```diff
@@ -132,6 +132,13 @@ func GetNodeBuildParameters(proj *sdk.Project, w *sdk.Workflow, runContext nodeR
 		delete(vars, "git.http.user")
 	}
 
+	if hookEvent != nil {
+		vars["parent.project"] = hookEvent.ParentWorkflow.Key
+		vars["parent.run"] = fmt.Sprintf("%d", hookEvent.ParentWorkflow.Run)
+		vars["parent.name"] = hookEvent.ParentWorkflow.Name
```
[the remark this answers is not in the document]
Comment (yesnault) [reply engine/api/workflow/process_parameters.go]: done

**Pull request merged**